packages/internal/src/lib-chainStorage.js

```diff
@@ -73,51 +83,57 @@ harden(assertPathSegment);
  */
 
 /**
- * Create a root storage node for a given backing function and root path.
- *
- * @param {(message: StorageMessage) => any} handleStorageMessage a function for sending a storageMessage object to the storage implementation (cf. golang/cosmos/x/vstorage/vstorage.go)
- * @param {string} rootPath
- * @param {object} [rootOptions]
- * @param {boolean} [rootOptions.sequence] employ a wrapping structure that preserves each value set within a single block, and default child nodes to do the same
+ * @param {import('@agoric/zone').Zone} zone
  */
-export function makeChainStorageRoot(
-  handleStorageMessage,
-  rootPath,
-  rootOptions = {},
-) {
-  assert.typeof(rootPath, 'string');
-
+export const zoneChainStorageNode = zone => {
   /**
+   * @param {import('./callback').Callback} messenger
    * @param {string} path
    * @param {object} [options]
    * @param {boolean} [options.sequence]
    * @returns {StorageNode}
    */
-  function makeChainStorageNode(path, options = {}) {
-    const { sequence = false } = options;
-    const node = {
+  const makeChainStorageNode = zone.exoClass(
+    'ChainStorageNode',
+    ChainStorageNodeI,
+    /**
+     * @param {import('./callback').Callback} messenger
+     * @param {string} path
+     * @param {object} [options]
+     * @param {boolean} [options.sequence]
+     */
+    (messenger, path, { sequence = false } = {}) =>
+      harden({ path, messenger, sequence }),
```

Comment:
Has this lost enforcement that the root path must be a string?
```suggestion
    (messenger, path, { sequence = false } = {}) => {
      assert.typeof(path, 'string');
      return harden({ path, messenger, sequence })
    },
```